### ISSUE TITLE
Syntax highlighting for TSX files

### ISSUE THREAD

**Issue by ubmit** [nvim-lua/kickstart.nvim#136]
**Describe the bug**
The syntax highlighting for TSX isn't working properly. You can take a look at the color of the HTML tags and the content inside them, they are the same color. I'm using the same theme settings that are in the `init.lua` repo. However, I've also tried another theme, the Catppuccin, and I faced inconsistencies with the syntax highlighting as well.

<img width="1640" alt="image" src="https://user-images.githubusercontent.com/8797405/212774133-538910dd-798f-4526-a8a4-0a7c12c261aa.png">

**To Reproduce**
1. Follow the guide in the README
2. Open a TSX file

**Desktop (please complete the following information):**
- OS: macOS Ventura 13.1
- Terminal: Kitty 0.26.4

** Neovim Version **
 - Output of running `:version` from inside of neovim: NVIM v0.8.2 
 
 
File example

```
import React from "react";

type Props = { text: string; color?: string };

function Title({ text, color }: Props) {
  const style = { color: color ?? "blueviolet" };
  return <h1 style={style}>{text}</h1>;
}

const Header = () => <header>Header component</header>;

export const App = () => (
  <>
    <Header />
    <Title text="Hello World" />
    <p>
      lorem ipsum dolor etc lorem ipsum dolor etc lorem ipsum dolor etc lorem
      ipsum dolor etc lorem ipsum dolor etc lorem ipsum dolor etc
    </p>
  </>
);
```

**Comment by ubmit** [comment]
I had to install `tsx` (https://github.com/tree-sitter/tree-sitter-typescript) for tree-sitter. Since this is a popular syntax people may want to have built-in and `typescript` is already supported in the `ensure_installed` array, I think it would be nice to have `tsx` there by default as well. What do you think? I'll make a PR for it :+1:

**Comment by tjdevries** [comment]
Thanks, sorry for the late reply. I will add `tsx` by default in #178

**Comment by ubmit** [comment]
Closed in favor of #178
